src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -475,12 +479,58 @@ ReturnRowsFromTuplestore(uint64 tupleCount, TupleDesc tupleDescriptor,
  * connection so that we receive results a row at a time.
  */
 static bool
-SendQueryInSingleRowMode(PGconn *connection, char *query)
+SendQueryInSingleRowMode(PGconn *connection, char *query, ParamListInfo paramListInfo)
 {
 	int querySent = 0;
 	int singleRowMode = 0;
 
-	querySent = PQsendQuery(connection, query);
+	if (paramListInfo != NULL)
+	{
+		int parameterIndex = 0;
```

Comment:
Can we put the contents of this in a separate function? It seems likely that we'll need it in other places too, and it's quite reasonably standalone.